### PR TITLE
docs: create overall architecture

### PR DESCRIPTION
**Pull request by anxolin**
# Summary

Moves the diagram from:
https://github.com/cowprotocol/cowswap-diagrams/pull/1

The idea is to have it closer to the code. This way we can maintain easily the module diagrams, and the overall. 

https://github.com/cowprotocol/cowswap-diagrams will contain broader diagrams or initiatives and RFCs.

# To Test
- Review diagram 
- Review readme (let me have some TODOs there, as I will want to do more PRs soon)

https://github.com/cowprotocol/cowswap/blob/overall-architecture/docs/architecture-overview.md

### CODE REVIEW

**Review by shoom3301 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -182,6 +182,7 @@ The plan:
 
 `emergency.js` is not cached by browser and loaded before all.
 
-## Nested docs
+## Documentation
 
-1. [Amounts formatting](src/cow-react/utils/amountFormat/README.md)
+1. [Oveall Architecture](./docs//architecture-overview.md)
```
Comment (alfetopito) [inline README.md:187]: Double slashes

```suggestion
1. [Oveall Architecture](./docs/architecture-overview.md)
```

File: docs/architecture-overview.drawio.svg
Comment (alfetopito) [inline docs/architecture-overview.drawio.svg:1]: Does the current project structure already match the diagram?
Or does it require code changes to look like this.
For example the `misc` module

Comment (anxolin) [reply docs/architecture-overview.drawio.svg:1]: It doesn't. There's a matching README on some changes I suggest. 

My idea is to propose those changes in PRs if everyone agrees

File: README.md (hunk 1 above)
Comment (anxolin) [reply README.md]: I saw it, and ammended the commit. but I never pushed the changes. fixed!

File: docs/architecture-overview.drawio.svg
Comment (anxolin) [reply docs/architecture-overview.drawio.svg:1]: There's a mention to this in the diagram (that is the coming from the common module) so i hope is fine. 

I also use different colors for the modules not yet implemented, so hopefully is fine.

**Comment by anxolin** [comment]
Regarding module names do you think is fine, or is this something we should discuss a bit more?

**Review by nenadV91 — APPROVED** [review]

**Review by alfetopito — APPROVED** [review]

**Pull request merged**